### PR TITLE
JAMES-2347 Demonstrate that IsOverQuota matcher is working in mailet …

### PR DESCRIPTION
**Pull request by aduprat**
…pipeline

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    @After
+    public void tearDown() throws Exception {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void aBounceMessageShouldBeSentToTheSenderWhenRecipientAsReachedHisQuota() throws Exception {
+        QuotaProbe quotaProbes = jamesServer.getProbe(QuotaProbesImpl.class);
+        quotaProbes.setMaxStorage(RECIPIENT_QUOTA_ROOT, new SerializableQuotaValue<>(SMALL_SIZE));
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java]: Gotcha, found an escaping "mismatch" in webadmin urls for user quotas... Restassured escapes '@'

**Comment by aduprat** [comment]
I don't understand the link between mailet integration tests and the other commits.

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aduprat) [reply server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java]: We are not testing WebAdmin, but mailets; I think that QuotaProbe is simpler in our case.

**Comment by chibenwa** [comment]
- https://github.com/linagora/james-project/pull/1315/commits/32f137424557dda3a1ae4916ff0fac3cbc6b8767 Solving warning. We tend to ship it as part of other PRs without worrying.
 - https://github.com/linagora/james-project/pull/1315/commits/89d93aac13fa3f46baa236795c2d588fe7c688f7 made my tests fails
 - https://github.com/linagora/james-project/pull/1315/commits/4b12e6267706f6145338a10a2028f935399e7a76 Missing binding so I could not use WebAdmin routes in integration tests
 - https://github.com/linagora/james-project/pull/1315/commits/7979c63cb187edf70d4916ebc4d75e319291bfd4 Use webadmin in integration tests... Rather explicit
 - https://github.com/linagora/james-project/pull/1315/commits/2d858a07cc75565deb9b6d96593bf797908eefaa Add integration tests for QuotaCount is also rather explicit...

**Comment by aduprat** [comment]
Can't approve mine, but you have my :+1:

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserQuotaRoutes.java [hunk 2]
```diff
@@ -61,7 +65,7 @@
 @Produces("application/json")
 public class UserQuotaRoutes implements Routes {
 
-    private static final String USER = "user";
+    public static final String USER = "user";
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserQuotaRoutes.java]: you shouldn't do that

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserQuotaRoutes.java [hunk 3]
```diff
@@ -248,8 +252,11 @@ public void defineGetQuotaCount() {
         }, jsonTransformer);
     }
 
-    private User checkUserExist(Request request) throws UsersRepositoryException {
-        String user = request.params(USER);
+    private User checkUserExist(Request request) throws UsersRepositoryException, UnsupportedEncodingException {
+        String user = URLDecoder.decode(request.params(USER),
+            StandardCharsets.UTF_8.displayName());
+
+        LoggerFactory.getLogger("any").error("user : " + user);
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserQuotaRoutes.java]: debug ?

File: server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserQuotaRoutesTest.java [hunk 4]
```diff
@@ -138,6 +139,15 @@ public void putCountShouldReturnNotFoundWhenUserDoesntExist() {
             .statusCode(HttpStatus.NOT_FOUND_404);
     }
 
+    @Test
+    public void putCountShouldAcceptEscapedUsers() {
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserQuotaRoutesTest.java]: it's not specific to put count

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java [hunk 5]
```diff
@@ -140,4 +141,39 @@ public void aBounceMessageShouldBeSentToTheRecipientWhenRecipientQuotaIsNotExcee
             .select(IMAPMessageReader.INBOX)
             .awaitMessage(awaitAtMostOneMinute);
     }
+
+    @Test
+    public void aBounceMessageShouldBeSentToTheSenderWhenRecipientAsReachedHisCOUNTQuota() throws Exception {
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/IsOverQuotaMatcherTest.java]: why COUNT and not Count ?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**